pkg/machineconfig/in_use_machineconfigs.go

```diff
@@ -0,0 +1,58 @@
+package machineconfig
+
+import (
+	"context"
+	"fmt"
+
+	mcfgclientset "github.com/openshift/client-go/machineconfiguration/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// GetInUseMachineConfigs filters in-use MachineConfig resources and returns set of their names.
+func GetInUseMachineConfigs(ctx context.Context, clientConfig *rest.Config, poolFilter string) (sets.Set[string], error) {
```

Comment:
Optionally I can pass the client instances (instead of the config). I think it would allow for some testing.